### DISCUSSION TITLE
Multiple Installed Tunnels for the same network segments

### DISCUSSION THREAD

**Discussion by erickhdz16** [strongswan/strongswan#1475] [Q&A]
Is it normal to have multiple installed tunnels form the same network segments??

### See how ipsec status to-client looks like:

 to-client[187]: ESTABLISHED 55 minutes ago, 172.31.1.177[54.208.101.182]...165.98.53.22[165.98.53.22]
  to-client[187]: IKEv2 SPIs: 59e01a29bcc5589c_i* de2a9b53246ab020_r, pre-shared key reauthentication in 113 minutes
  to-client[187]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  to-client{811}:  INSTALLED, TUNNEL, reqid 3, ESP SPIs: cbde6860_i 2a6b2036_o
  to-client{811}:  AES_CBC_256/HMAC_SHA2_256_128, 21137 bytes_i (290 pkts, 2s ago), 41363 bytes_o (564 pkts, 2s ago), rekeying in 34 minutes
  **to-client{811}:   172.31.0.96/29 === 10.218.30.240/28**
  to-client[186]: ESTABLISHED 60 minutes ago, 172.31.1.177[54.208.101.182]...165.98.53.22[165.98.53.22]
  to-client[186]: IKEv2 SPIs: d6d7e9c1f33ca77f_i* 70587778db83474f_r, pre-shared key reauthentication in 105 minutes
  to-client[186]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  to-client{808}:  INSTALLED, TUNNEL, reqid 3, ESP SPIs: ca151083_i 2a6b1c0a_o
  to-client{808}:  AES_CBC_256/HMAC_SHA2_256_128, 8529 bytes_i (119 pkts, 2s ago), 16716 bytes_o (230 pkts, 2s ago), rekeying in 25 minutes
  **to-client{808}:   172.31.0.96/29 === 10.218.30.240/28**
  to-client{810}:  INSTALLED, TUNNEL, reqid 3, ESP SPIs: cb59c6db_i 2a6b1e1a_o
  to-client{810}:  AES_CBC_256/HMAC_SHA2_256_128, 8724 bytes_i (122 pkts, 2s ago), 18722 bytes_o (238 pkts, 2s ago), rekeying in 28 minutes
  **to-client{810}:   172.31.0.96/29 === 10.218.30.240/28**

### Current vpn config:

conn to-client
  authby=secret
  leftid=zzzzzzzz
  left=yyyyyyyy
  leftsubnet=172.31.0.96/29
  right=xxxxxxxxx
  rightsubnet=10.218.30.240/28
  auto=route
  keyexchange=ikev2
  ike=aes256-sha256-modp2048
  esp=aes256-sha256
  dpddelay=30
  dpdtimeout=120
  dpdaction=restart

### strongswan.conf:

# strongswan.conf - strongSwan configuration file
#
# Refer to the strongswan.conf(5) manpage for details
#
# Configuration changes should be made in the included files

charon {
        load_modular = yes
        plugins {
                include strongswan.d/charon/*.conf
        }
}

include strongswan.d/*.conf

**Comment by tobiasbrunner** [comment]
Depends on a lot of things (in particular the strongSwan version). But with `auto=route` and `dpdction=restart` (as well as `reauth=yes`, which is the default when configured with ipsec.conf and break-before-make reauthentication, also the default, see [here](https://docs.strongswan.org/docs/5.9/config/rekeying.html) for details), it could very well happen with older releases. When the old SA is deleted and recreated, concurrent acquires from the kernel (due to `auto=route`) could trigger duplicate CHILD_SAs.

**Reply by erickhdz16** [reply]
Hi Tobias, thanks for your answer.

### Extra info:
I have two more vpn's configurations in the same strongswan instance using the same configuration (just different names and ip's) and this behavior only occurs in the vpn mentioned before.
The version is **Linux strongSwan U5.9.5/K5.15.0-1026-aws**

All the configurations for ipsec.conf and charon are default values. 

The duplication of "INSTALLED" tunnels does not happen if I turn down and up the vpn. After reading the link provided it looks like it only happens during rekeying.

Would you suggest to change **auto** param to other value like **start**?

**Reply by tobiasbrunner** [reply]
> After reading the link provided it looks like it only happens during rekeying.

No, during reauthentication (different concept than rekeying) or if the connection is deleted due to DPDs. Check the log for what exactly happens and what triggers the SAs.

> Would you suggest to change **auto** param to other value like **start**?

Depends on your use case and situation. For instance, if there is traffic that could initiate the connection from this host via acquire after the connection was closed due to DPDs (and also initially). In which case using `auto=route` with `dpdaction=clear` and `reauth=no` would be the combination of choice. Unless you require reauthentication instead of rekeying and can't use make-before-break reauthentication. In that case, using `start/restart` would be the better choice with older releases (before [5.9.6](https://github.com/strongswan/strongswan/releases/tag/5.9.6)). That could also be better if the traffic is primarily initiated by the other end and you can't use `auto=route` on both sides (note that the latter can lead to duplicate IKE and CHILD_SAs if both peers initiate the connection concurrently, but that shouldn't be a problem).
